color/src/color.rs

```diff
@@ -358,6 +371,19 @@ impl<CS: ColorSpace> AlphaColor<CS> {
             _ => self.map_in::<Oklab>(|l, a, b, alpha| [f(l), a, b, alpha]),
         }
     }
+
+    /// Map the hue of the color.
+    ///
+    /// In a color space that naturally has a hue component, map that value.
+    /// Otherwise, do the mapping in [Oklch]. The hue is in degrees.
+    #[must_use]
+    pub fn map_hue(self, f: impl Fn(f32) -> f32) -> Self {
+        match CS::LAYOUT {
+            ColorSpaceLayout::HueFirst => self.map(|h, c1, c2, a| [f(h), c1, c2, a]),
+            ColorSpaceLayout::HueThird => self.map(|c0, c1, h, a| [c0, c1, f(h), a]),
+            _ => self.map_in::<Oklch>(|l, c, h, alpha| [l, c, f(h), alpha]),
+        }
```

Comment:
To reduce duplication, this could be

```rust
let (opaque, alpha) = self.split();
opaque.map_hue(f).with_alpha(alpha)
```

The same is true for `AlphaColor::map_lightness`; I copied that style.